code/obj/item/clothing/hats.dm

```diff
@@ -959,6 +959,24 @@ TYPEINFO(/obj/item/clothing/head/that/gold)
 		..()
 		src.color = "#FF8800"
 
+/obj/item/clothing/head/beret/dyeable
+	name = "dyeable beret"
+	desc = "Can be dyed with hair dye. Obviously."
+	icon_state = "beret_base"
+	item_state = "dye_beret"
+
+	New()
+		..()
+		src.color = "#FFFFFF"
+
+	attackby(obj/item/dye_bottle/W, mob/user)
+		. = ..()
+		if (istype(W, /obj/item/dye_bottle))
+			src.color = W.customization_first_color
+			src.UpdateIcon()
+			user.update_clothing()
+			user.visible_message(SPAN_ALERT("<b>[user]</b> splashes dye on [his_or_her(user)] beret."))
+
```

Comment:
A couple of nitpicks with this:
- The code should return and not call parent if the conditional succeeds, otherwise it'll play your message and "user hits the beret with the dye bottle"
- You can't assume that `user` is always going to be the wearer of the beret, since you can use items on things other people are wearing. It'd be better to check if `src.loc` is a mob and then call `update_clothing` on that if it is.

Same applies to the sweater too.